tools/migrators/lib/grafana/service_model_client.py

```diff
@@ -0,0 +1,201 @@
+    This uses the k8s API to interact with the service model which is implemented
+    as a Kubernetes ApiServer embedded within Grafana.
+    """
+
+    @staticmethod
+    def parse_k8s_url(url: str) -> tuple:
+        """
+        Parse a kubernetes URL of the format https://<namespace>:<token>@<server>
+        Returns tuple of (server_url, namespace, token)
+        """
+        parsed = urlparse(url)
+        if not all([parsed.scheme, parsed.netloc]):
+            raise ValueError(
+                "Invalid URL format. Expected: https://<namespace>:<token>@<server>"
+            )
+
+        # Split username (namespace) and password (token)
+        if "@" not in parsed.netloc:
+            raise ValueError(
+                "URL must contain credentials in the format namespace:token@server"
+            )
+
+        auth, server = parsed.netloc.rsplit("@", 1)
+        if ":" not in auth:
+            raise ValueError("Credentials must be in the format namespace:token")
+
+        namespace, token = auth.split(":", 1)
+
+        # Reconstruct server URL with scheme
+        server_url = f"{parsed.scheme}://{server}{parsed.path}"
+
+        return server_url, namespace, token
+
+    def __init__(self):
+        """
+        Initialize the ServiceModelClient.
+        Configures the client using a URL-based format or falls back to legacy configuration.
+        """
+        if GRAFANA_SERVICE_ACCOUNT_URL:
+            try:
+                server_url, namespace, token = self.parse_k8s_url(
+                    GRAFANA_SERVICE_ACCOUNT_URL
+                )
+
+                # Configure client using parsed parameters
+                configuration = client.Configuration()
+                configuration.host = server_url
+                configuration.api_key = {"authorization": f"Bearer {token}"}
+                # configuration.verify_ssl = False  # Note: In production, you should handle SSL verification properly
+
+                # Set the default namespace
+                self.default_namespace = namespace
+
+                # Create API client with custom configuration
+                client.Configuration.set_default(configuration)
+                self.api_client = client.ApiClient(configuration)
+
+            except ValueError as e:
+                raise ValueError(
+                    f"Failed to parse GRAFANA_SERVICE_ACCOUNT_URL: {str(e)}"
+                )
+        else:
+            # Try to load default kubeconfig
+            try:
+                config.load_kube_config()
+                self.api_client = client.ApiClient()
+                self.default_namespace = "default"
+                print("Loaded KUBECONFIG")
+            except config.config_exception.ConfigException:
+                raise ValueError(
+                    "Unable to configure Kubernetes client. Please set: "
+                    "GRAFANA_SERVICE_ACCOUNT_URL (format: https://<namespace>:<token>@<server>) "
+                )
```

Comment:
I would consider dropping this and forcing the user to just always specify `GRAFANA_SERVICE_ACCOUNT_URL` (this seems rather implicit/undocumented)